### PR TITLE
Zendesk connector - UI

### PR DESCRIPTION
**Pull request by albandum**
## Description

This PR drafts a simple version of the Zendesk connector oauth. The actual connector will likely start as a labs alpha feature (will need to show in the UI). This allows getting started on the Zendesk oauth flow and showing it to the users as we did with microsoft - we'll keep a list of users asking for Zendesk beta testing. 

- This PR handles the UI 
- This PR handles the oAuth flow (dependent on https://github.com/dust-tt/dust/pull/6589)
- oAuth flow is gated behind FF - preview for the rest

Screenshot 2024-07-29 at 19 46 45

https://github.com/user-attachments/assets/fca35211-ada5-489c-bdf7-f4122bc85f45

## Deploy Plan
https://github.com/dust-tt/dust/pull/6589 needs to be merged first
Once this is live we can start asking Zendesk for a *global oAuth client* - we will have to go through a review process so better to do it early on

### CODE REVIEW

**Comment by spolu** [comment]
Just to be clear, but you probably already have this. The idea is to create something entirely out of connectors at first. So no code committed expect maybe one script run manually daily somewhere.

All that being said, fine to show it as preview.

**Comment by albandum** [comment]
> Just to be clear, but you probably already have this. The idea is to create something entirely out of connectors at first. So no code committed expect maybe one script run manually daily somewhere.

Ha, ok, was going for something a little more solid. Will think about a light way to do this then but it feels like the path to a regular connector isn't too long

File: front/components/assistant_builder/shared.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const CONNECTOR_PROVIDER_TO_RESOURCE_NAME: Record<
   github: { singular: "repository", plural: "repositories" },
   intercom: { singular: "element", plural: "elements" },
   webcrawler: { singular: "page", plural: "pages" },
+  zendesk: { singular: "ticket", plural: "tickets" },
```
Comment (flvndvd) [inline front/components/assistant_builder/shared.ts:33]: Is `ticket` the only available resource on Zendesk?

File: front/lib/api/oauth.ts [hunk 2]
```diff
@@ -216,6 +216,25 @@ const PROVIDER_STRATEGIES: Record<
       return getStringFromQuery(query, "state");
     },
   },
+  zendesk: {
+    setupUri: (connection) => {
+      const scopes = ["tickets:read", "hc:read"];
```
Comment (flvndvd) [inline front/lib/api/oauth.ts:221]: Shouldn't those scopes be aligned with the one in zendesk.rs?

File: front/package-lock.json [hunk 3]
```diff
@@ -10736,9 +10736,9 @@
       "license": "Apache-2.0"
     },
     "node_modules/@dust-tt/sparkle": {
-      "version": "0.2.193",
-      "resolved": "https://registry.npmjs.org/@dust-tt/sparkle/-/sparkle-0.2.193.tgz",
-      "integrity": "sha512-ehUgBTsBv6kqZD/eWxeZ7CD5fMRBBSrkddCvs6YMI+QxD1yhpLut4mc9aJ0XMreVx6FwHnQCoqkWMisgZtT4fQ==",
+      "version": "0.2.195",
```
Comment (flvndvd) [inline front/package-lock.json:10739]: Weird that you bump only the package-lock.json, expected?

**Review by flvndvd — APPROVED** [review]
LGTM, left a few questions.

File: front/package-lock.json (hunk 3 above)
Comment (albandum) [reply front/package-lock.json:10739]: Yeah, package was supposedly bumped there https://github.com/dust-tt/dust/pull/6577/files#diff-1335400798ac3f1df5513672e631b9269f8febecf18045d2dd4da05afcbe81c9 but forgot to bump front 🤷

File: front/lib/api/oauth.ts (hunk 2 above)
Comment (albandum) [reply front/lib/api/oauth.ts:221]: Indeed - mb in zendesk.rs, I reverted a change. Adding the fix in this PR

File: front/components/assistant_builder/shared.ts (hunk 1 above)
Comment (albandum) [reply front/components/assistant_builder/shared.ts:33]: No articles as well - I don't see a way of adding both wordings?

File: front/package-lock.json (hunk 3 above)
Comment (flvndvd) [reply front/package-lock.json:10739]: Yeah but then it should bump the pacakge.json as well, no?

File: front/components/assistant_builder/shared.ts (hunk 1 above)
Comment (flvndvd) [reply front/components/assistant_builder/shared.ts:33]: We went with element/element for Intercom we can probably align?

**Comment by spolu** [comment]
Nope no OAuth integration for now as it would commit us to building it to some extent and would require framing etc...

This should happen entirely outside of the product / infra for now. The easiest way is to have users create the credentials without oAuth (generally available as developer).

Basically the rule of engagement here is: no code in any of our systems.

To be honest this PR is not needed IMHO. Let's unblock a first user and learn with them and take it from there.

**Comment by spolu** [comment]
Rationale being that integrating it in oAuth means long term support to which we are not committed yet

**Comment by flvndvd** [comment]
> Rationale being that integrating it in oAuth means long term support to which we are not committed yet

I'm not sure to understand the reasoning here. We haven't committed or communicated on any dates, it looks like we're gauging interest in supporting Zendesk as a data source in the future. Why not fast-track things by leveraging our existing bricks? FWIW, I really believe that Gong should have been integrated into our connectors stack. Currently, there are quite a few errors slipping through our monitors, but some customers are still using it, and they deserve the same level of support and SLA.

**Comment by albandum** [comment]
I don't see how this would differ being oAuth or not - with the right [alpha] stamps or a specific page explaining that this is a test connector, how would that be different? 

Not sure how you envision that tryout then 🤔 With an API key connection?

**Comment by spolu** [comment]
The goal for now is for the code to run in our users infra not ours. @albandum supporting them achieving that.

This is the goal for now.

With that in mind. There is obviously no oAuth support as it would mean exfilling tokens from our infra which is an anti pattern.

**Comment by spolu** [comment]
If the goals are unclear let's just pause this for now and revert 👍

**Comment by spolu** [comment]
Solutions is not an alternative to the engineering prioritisation. 

The idea here is: "Zendesk is not supported but let me help you build a prototype integration on your end that pushes the data to Dust pending official support"

Sorry it it wasn't clear. In any case that means 0 commit to our codebase for now.

Well learn from it and that's great. So it still has value on top of unblocking users willing to go down that route.

**Comment by albandum** [comment]
Ok, not sure to understand the rationale there - in that case everything's doable through the API : transcripts too but we did  build it internally for user's comfort (didn't ask them to run a script to detect transcripts and call the API). Not sure to see how this is different.

We have a lot of Zendesk demand, I understand it got deprio but asking  the customer to build their own data extract doesn't seem like a great XP for them

But OK, let's pause!

**Comment by spolu** [comment]
Yes everything can be done through the API including transcripts.

We have a lot of demand yes and we do take that into account when prioritizing the Zendesk connector.

It's currently not prioritized right? So no eng resources should go to it. What your doing here is eng resources. It's building against production systems which will require maintenance, migration and monitoring. More future eng resources.

Now if we assume we don't allocate eng resources to this, we can still help our users build against our API. This is the goal of exploring Zendesk with them. Well learn from that and we'll also educate our users. It's still positive even if its does not scale. That's on purpose. Scaling is product which requires eng resources.

Does that make sense?

**Comment by spolu** [comment]
If you think it's not worth it (I strongly believe otherwise for transparency) let's pause indeed.

If we pause let's revert all code (does not mean it's dead it just means it's not run in production increasing the surface of one of our most critical system, the security argument is indiscutable here. You don't want a leak to be cause by dead Zendesk code right?)

**Comment by spolu** [comment]
If we don't pause let's also revert since this oAuth code will still be unused.

**Comment by albandum** [comment]
Cancelling until Zendesk connector gets reprio - switching to API solution 
Will revert https://github.com/dust-tt/dust/pull/6589

**Comment by albandum** [comment]
>  You don't want a leak to be cause by dead Zendesk code right?)

Def not :) 
Closing the PR and will revert https://github.com/dust-tt/dust/pull/6589

**Pull request closed without merging**